### DISCUSSION TITLE
Deprecation messages

### DISCUSSION THREAD

**Discussion by qt-thomas** [gdsfactory/gdsfactory#2526] [Q&A] [answered]
Hello. I have updated gdsfactory and now get a few deprecation messages. Can you explain to me how to implement the functionality that is being deprecated? I give a few examples below. As a reward, I promise to submit a PR with a more descriptive warning message suggesting what to do.

- decorator is deprecated and will be removed soon.
- metadata_child is deprecated and will be removed in future versions of gdsfactory
- prefix is deprecated and will be removed soon.

**Comment by joamatab** [comment]
Hi Thomas

We are deprecating all the cell magic methods 

- decorator. You can add a post process function to the pcells. See open PR
- name. You need to create a new cell with no parameters (fixed cell)
- prefix: same as name 
- metadata child: use info instead and cell_with_child decorator

**Reply by qt-thomas** [reply]
Wow thanks for the quick reply. Can you give me an example of specializing a gf.cell with gf.partial while specifying a new name or prefix?

**Reply by joamatab** [reply]
You can use a fixed cell for fixing the name


```python
import gdsfactory as gf

xf_sc = partial(gf.cross_section, radius=10, min_bend_radius=5)

@gf.cell
def straight_thomas():
     return gf.c.straight(cross_section=xf_sc)
```

You can still use it as a PCell with the new `straight_thomas` prefix
```python
@gf.cell
def straight_thomas(**kwargs):
     return gf.c.straight(cross_section=xf_sc, **kwargs)
```

**Reply by qt-thomas** [reply]
Ok so we should stop specializing cells with partial and use native functions.

**Reply by joamatab** [reply]
if you want to control the names, yes

you can still use partials when you don't care about getting specific names

**Reply by nikosavola** [reply]
As the _fixed cell_ method here seems to be the go-to one, it should be better indicated in the docs I think.

**Comment by joamatab** [comment, marked as answer]
```python

    ##################
    # Deprecated, using decorator that modifies the component
    ##################
    # c = gf.components.straight(decorator=gf.add_padding)

    ##################
    # Deprecated, using decorator that returns a new component (container)
    ##################
    # c = gf.components.straight(decorator=gf.routing.add_fiber_array)

    ##################
    # now, using post_process that modifies the component
    ##################
    c = gf.components.straight(post_process=gf.add_padding)

    ##################
    # now, using a function that returns a new component (container)
    ##################
    c = gf.components.add_fiber_array(gf.components.straight())
    c.show()
```

**Reply by qt-thomas** [reply]
Looks like this post_process is a feature available in the next version of gdsfactory, correct? Maybe a minor release, 7.12?

**Comment by qt-thomas** [comment]
By the way. Checkout this idea for gf.cell: https://stackoverflow.com/questions/74074580/how-to-avoid-losing-type-hinting-of-decorated-function

**Reply by nikosavola** [reply]
I think I attempted the `ParamSpec` approach but couldn't quite get it working. You should try if type checking this advanced is more familiar to you 😃
